docs/fundamentals/networking/telemetry/metrics.md

```diff
@@ -0,0 +1,193 @@
+The [`http.client.request.duration`](../../../core/diagnostics/built-in-metrics-system-net.md#instrument-httpclientrequestduration) instrument supports enrichment by registering callbacks with the <xref:System.Net.Http.Metrics.HttpMetricsEnrichmentContext>.
+Note that this is a low-level API and a separate callback registration is needed for each `HttpRequestMessage`.
+
+A simple way to do the callback registration at a single place is to implement a custom delegating <xref:System.Net.Http.DelegatingHandler>.
+This will allow you to intercept and modify the requests before they are forwarded to the inner handler and sent to the server:
+
+:::code language="csharp" source="snippets/metrics/Program.cs" id="snippet_Enrichment":::
+
+In case you are working with [`IHttpClientFactory`](../../../core/extensions/httpclient-factory.md), you can use xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A to register the `EnrichmentHandler`:
+
+:::code language="csharp" source="snippets/metrics/Program.cs" id="snippet_EnrichmentWithFactory":::
+
+> [!NOTE]
+> For performance reasons, the enrichment callback is only invoked when the `http.client.request.duration` instrument is enabled, meaning that something should be collecting the metrics.
+> This can be `dotnet-monitor`, Prometheus exporter, a [`MeterListener`](../../../core/diagnostics/metrics-collection.md#create-a-custom-collection-tool-using-the-net--api) or a `MetricCollector<T>`.
+
+## `IMeterFactory` and `IHttpClientFactory` integration
+
+HTTP metrics were designed with isolation and testability in mind meaning that metrics can be emitted by a custom <xref:System.Diagnostics.Metrics.Meter> instance.
+This can be achived by assigning a custom <xref:System.Diagnostics.Metrics.IMeterFactory> instance to <xref:System.Net.Http.SocketsHttpHandler.MeterFactory?displayProperty=nameWithType> or <xref:System.Net.Http.HttpClientHandler.MeterFactory?displayProperty=nameWithType>.
+
+When working with [`Microsoft.Extensions.Http`](https://www.nuget.org/packages/microsoft.extensions.http) and [`IHttpClientFactory`](../../../core/extensions/httpclient-factory.md), the default `IHttpClientFactory` implementation will automatically pick the `IMeterFactory` instance registered in the `ServiceCollection` and assign it to the primary handler it creates internally.
+
+> [!NOTE]
+> The <xref:System.Diagnostics.Metrics.Meter.Name?displayProperty=nameWithType> is `System.Net.Http` for all metrics emitted by `HttpClientHandler` / `SocketsHttpHandler`.
+
+### Testing metrics
+
+The following example demonstrates how to validate built-in metrics in unit tests using xUnit, `IHttpClientFactory` and `MetricCollector<T>` from the [`Microsoft.Extensions.Diagnostics.Testing`](https://www.nuget.org/packages/Microsoft.Extensions.Diagnostics.Testing) NuGet package:
+
+:::code language="csharp" source="snippets/metrics/Program.cs" id="snippet_Testing":::
+
+## Metrics vs. EventCounters
```

Comment:
I forgot to push this paragraph yesterday, @JamesNK @MihaZupan  can you take a look?

Otherwise addressed all feedback.